### DISCUSSION TITLE
Generating release notes with git log

### DISCUSSION THREAD

**Discussion by thevladyslav** [codemagic-ci-cd/codemagic-docs#2041] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Codemagic workflow triggers upon creating tag. So to generate release notes from git commits I have a script that does the following.

1. Takes previous tag
2. Generates release notes from previous tag to the most recent one, that triggered Codemagic workflow.

This is the script:

```
set -x

git fetch --all --tags


prev_tag=$(git for-each-ref --sort=-creatordate  --format '%(objectname)' refs/tags | sed -n 2p )
notes=$(git log --pretty=format:"\n- %s" "$prev_tag"..HEAD)

echo "$notes" | tee release_notes_en.txt
```
This script works perfectly on local machine, but for some reason it doesn't work on Codemagic's side.
If I want to generate release notes between tags _v1.0.0_ and _v2.0.0_
**This is what prints script when run locally:**

> v2.0.0
> 
> Fix bug A
> 
> Fix bug B

**This is what prints script when run on Codemagic**:

> v2.0.0

As you can see, Codemagic able to print only first commit, ignoring all others. This is happens only if workflow was triggered by tag. Even simple git log returns only one commit, and it's current commit, on which workflow is running (triggered by tag). 
If i run workflow manually from branch instead of tag, then result is ok.

### Expected results

Generate same result as on local machine

> v2.0.0
> 
> Fix bug A
> 
> Fix bug B

### Actual results

Codemagic able to print only first commit, ignoring all others

> v2.0.0

**Comment by dtrdic** [comment, marked as answer]
hey @personaljesusua, although we are chatting through Intercom, i'll post here as well..

The issue is related to the **CM_CLONE_DEPTH** environment variable, which is configured to clone only one commit by default for builds triggered by tags.

To generate release notes between tags, you may need to adjust the **CM_CLONE_DEPTH** environment variable to a higher value that suits your requirements. For example, if you want to capture all commits between tags v1.0.0 and v2.0.0, you can set CM_CLONE_DEPTH to a number greater than the number of commits between those tags (e.g., 10 or more). This change will allow your script to fetch and generate release notes from the desired commit range.

Keep in mind that modifying CM_CLONE_DEPTH might increase the time it takes to clone the repository during the build, so consider the trade-off between clone depth and build performance.
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/61195609-2956-4ff8-888f-49ea2a0735ee)
